src/common/ServerCompatBanner.js

```diff
@@ -0,0 +1,101 @@
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+const styles = createStyleSheet({
+  wrapper: {
+    backgroundColor: HALF_COLOR,
+    padding: 8,
+  },
+  textRow: {
+    padding: 8,
+    flexDirection: 'row',
+  },
+  buttonsRow: {
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+});
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying that the server version is unsupported.
+ *
+ * Currently just checks if it's less than 2.1.0, since that's the
+ * marker that was put down in zulip/zulip@d7d79b57b. In the future,
+ * this won't have to be so hard-coded; we may use a timer or
+ * something.
+ */
+// Made with somewhat careful attention to
+// https://material.io/components/banners. Please consult that before
+// making layout changes, and try to make them in a direction that
+// brings us closer to those guidelines.
+export default function ServerCompatBanner(props: Props) {
+  const dispatch = useDispatch();
+  const hasDismissedServerCompatNotice = useSelector(
+    state => getSession(state).hasDismissedServerCompatNotice,
+  );
+  const zulipVersion = useSelector(state => getActiveAccount(state).zulipVersion);
+  const realm = useSelector(state => getActiveAccount(state).realm);
+  const isAdmin = useSelector(getIsAdmin);
+
+  if (zulipVersion && !zulipVersion.isAtLeast('2.1.0') && !hasDismissedServerCompatNotice) {
+    return (
+      <SafeAreaView mode="padding" edges={['right', 'left']} style={styles.wrapper}>
+        <View style={styles.textRow}>
+          {isAdmin ? (
+            <Label
+              text={{
+                text:
+                  '{realm} is running Zulip Server {serverVersionRaw}, which is unsupported. Please upgrade your server as soon as possible.',
+                values: { realm: realm.toString(), serverVersionRaw: zulipVersion.raw() },
+              }}
+            />
+          ) : (
+            <Label
+              text={{
+                text:
+                  '{realm} is running Zulip Server {serverVersionRaw}, which is unsupported. Please contact your administrator about upgrading.',
+                values: { realm: realm.toString(), serverVersionRaw: zulipVersion.raw() },
+              }}
+            />
+          )}
+        </View>
+        <View style={styles.buttonsRow}>
+          <ZulipTextButton
+            leftMargin
+            label="DISMISS"
+            onPress={() => {
+              dispatch(dismissCompatNotice());
+            }}
+          />
+          <ZulipTextButton
+            leftMargin
+            label={isAdmin ? 'FIX NOW' : 'LEARN MORE'}
+            onPress={() => {
+              openLinkWithUserPreference(
+                'https://zulip.readthedocs.io/en/stable/overview/release-lifecycle.html#compatibility-and-upgrading',
+                store.getState,
+              );
+            }}
+          />
+        </View>
+      </SafeAreaView>
+    );
+  } else {
+    return null;
+  }
```

Comment:
nit: swap the order of "then" and "else", and make this an early return

That way the reader isn't in suspense about the other case while reading through the bulk of this function's code, and also the code doesn't get an extra level of indentation.

Also that way we can easily have two separate conditionals, one for the version check and one for `hasDismissedServerCompatNotice`. I think that may be a bit cleaner to read.